internal/patch/add.go

```diff
@@ -0,0 +1,154 @@
+package patch
+
+import (
+	"fmt"
+	f "github.com/elimity-com/scim/internal/filter"
+	"github.com/elimity-com/scim/schema"
+	"github.com/scim2/filter-parser/v2"
+	"strings"
+)
+
+func (v OperationValidator) ValidateAdd() error {
+	// The operation must contain a "value" member whose content specifies the value to be added.
+	if v.value == nil {
+		return fmt.Errorf("an add operation must contain a value member")
+	}
+
+	// If "path" is omitted, the target location is assumed to be the resource itself.
+	if v.path == nil {
+		return v.validateAddEmptyPath()
+	}
+
+	refAttr, err := v.getRefAttribute(v.path.AttributePath)
+	if err != nil {
+		return err
+	}
+	if v.path.ValueExpression != nil {
+		fmt.Println(v.path.ValueExpression, refAttr.Name())
+		if err := f.NewFilterValidator(v.path.ValueExpression, schema.Schema{
+			Attributes: refAttr.SubAttributes(),
+		}).Validate(); err != nil {
+			return err
+		}
+	}
+	if subAttrName := v.path.SubAttributeName(); subAttrName != "" {
+		refSubAttr, err := v.getRefSubAttribute(refAttr, subAttrName)
+		if err != nil {
+			return err
+		}
+		refAttr = refSubAttr
+	}
+
+	if refAttr.MultiValued() {
+		if list, ok := v.value.([]interface{}); !ok {
+			attr, err := refAttr.ValidateSingular(v.value)
+			if err != nil {
+				return err
+			}
+			v.value = []interface{}{attr}
+		} else {
+			var attrs []interface{}
+			for _, value := range list {
+				attr, err := refAttr.ValidateSingular(value)
+				if err != nil {
+					return err
+				}
+				attrs = append(attrs, attr)
+			}
+			v.value = attrs
```

Comment:
Why do we need these mutations?

Reply:
`ValidateSingular` removes unknown attributes from the given values.

Reply:
> `ValidateSingular` removes unknown attributes from the given values.

Would it require a lot of effort to avoid these mutations?